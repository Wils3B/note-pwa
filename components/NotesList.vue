<template>
  <v-row>
    <v-col
      v-for="(n, index) in notes"
      :key="index"
      cols="6"
      sm="4"
      md="3"
      lg="2"
      class="note-wrapper"
    >
      <app-note-card :note="n" />
    </v-col>
  </v-row>
</template>

<script>
import AppNoteCard from '~/components/NoteCard.vue'

export default {
  components: {
    AppNoteCard
  },
  props: {
    notes: {
      type: Array,
      default: () => []
    }
  }
}
</script>

<style scoped>
.note-wrapper {
  padding: 8px;
}
</style>
