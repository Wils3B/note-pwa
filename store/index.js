export const state = () => {
  return {
    colors: ['white', 'yellow', 'purple', 'green', 'red', 'blue', 'brown'],
  }
}
