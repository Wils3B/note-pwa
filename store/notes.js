import DB from '~/model/db'
import Note from '~/model/note'

export const state = () => {
  const notes = []
  return {
    notes,
  }
}

export const mutations = {
  saveNote(currentState, note) {
    currentState.notes.push(note)
    DB.addNotes(note)
  },
  deleteNote(currentState, id) {
    const note = currentState.notes.filter((note) => note.id === id)[0]
    const index = currentState.notes.indexOf(note)
    currentState.notes.splice(index, 1)
    DB.deleteNotes(id)
  },
  editNote(currentState, note) {
    const index = currentState.notes.indexOf(
      getters.noteById(currentState)(note.id)
    )
    const currentNote = currentState.notes[index]
    currentNote.color = note.color
    Note.setContent(currentNote, note.content)
    currentState.notes.splice(index, 1)
    currentState.notes.push(currentNote)
    DB.putNotes(note)
  },
  setAllNotes(currentState, notes) {
    currentState.notes = notes
  },
}

export const getters = {
  noteById(currentState) {
    window.notes = currentState.notes
    return (id) => currentState.notes.filter((note) => note.id === id)[0]
  },
}

export const actions = {
  async fetchAllNotes(context) {
    if (!DB.connected) await DB.connect()
    let notes
    if (DB.connected) {
      notes = await DB.getAllNotes()
      context.commit(
        'setAllNotes',
        notes.data.sort((a, b) => (a.modifiedAt < b.modifiedAt ? -1 : 1))
      )
    }
  },
}
