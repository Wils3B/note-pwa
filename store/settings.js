export const state = () => {
  const loadedSettings = loadSettings()
  return {
    selecteds: {
      sortType: loadedSettings.sortType || 'date'
    },
    groups: {
      sortTypes: ['date', 'color']
    }
  }
}

export const mutations = {
  changeSortType(currentState, type) {
    currentState.selecteds.sortType = type
    storeSettings(currentState.selecteds)
  }
}

function storeSettings(settings) {
  localStorage.setItem('settings', JSON.stringify(settings))
}

function loadSettings() {
  const settingsJSON = localStorage.getItem('settings')
  let settingsObj = {}
  try {
    settingsObj = JSON.parse(settingsJSON)
  } catch (e) {}
  return settingsObj || {}
}
