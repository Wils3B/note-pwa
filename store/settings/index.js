import { loadSettings, storeSettings } from './functions'

export const state = () => {
  const loadedSettings = loadSettings()
  return {
    selecteds: {
      sortType: loadedSettings.sortType || 'date',
      defaultColor: loadedSettings.defaultColor || 'white',
      appFont: loadedSettings.appFont || 'Amita',
      autosave: loadedSettings.autosave || '2min',
    },
    groups: {
      sortTypes: ['date', 'color'],
      fonts: ['Kalam', 'Amita', 'Roboto'],
      autosave: ['off', '10s', '20s', '30s', '2min', '5min'],
    },
  }
}

export const mutations = {
  changeSortType(currentState, type) {
    currentState.selecteds.sortType = type
    storeSettings(currentState.selecteds)
  },
  changeDefaultColor(currentState, color) {
    currentState.selecteds.defaultColor = color
    storeSettings(currentState.selecteds)
  },
  changeAppFont(currentState, font) {
    currentState.selecteds.appFont = font
    storeSettings(currentState.selecteds)
  },
  changeAutosave(currentState, autosave) {
    currentState.selecteds.autosave = autosave
    storeSettings(currentState.selecteds)
  },
}

export const getters = {
  autosaveInMillis(currentState) {
    const setting = currentState.selecteds.autosave
    if (/^\d+s$/.test(setting)) {
      return 1000 * Number(setting.substring(0, setting.length - 1))
    } else if (/^\d+min$/.test(setting)) {
      return 1000 * 60 * Number(setting.substring(0, setting.length - 3))
    }
    return 0
  },
}
