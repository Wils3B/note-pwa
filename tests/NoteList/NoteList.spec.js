import { mount } from '@vue/test-utils'
import Vuetify from 'vuetify'
import { localInstance } from '../common'
import router from '../router'
import store from '../store'
import notes from '../notes'
import NotesList from '~/components/NotesList.vue'

const localVue = localInstance()

describe('Note List', () => {
  const wrapper = mount(NotesList, {
    localVue,
    router,
    store,
    vuetify: new Vuetify(),
    propsData: {
      notes,
    },
  })

  test('it should be a vue instance', () => {
    expect(wrapper.isVueInstance).toBeTruthy()
  })

  test('Have a correct render', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('it should have 12 note cards', () => {
    expect(wrapper.findAll('.note-list>.note-wrapper').length).toBe(12)
  })

  test('it should not have any card if any note is passed as prop', () => {
    const localWrapper = mount(NotesList, {
      localVue,
      router,
      store,
    })

    expect(localWrapper.findAll('.note-list>.note-wrapper').length).toBe(0)
  })
})
