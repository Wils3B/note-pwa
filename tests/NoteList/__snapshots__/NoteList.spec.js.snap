// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note List Have a correct render 1`] = `
"<div class=\\"row note-list\\">
  <div class=\\"note-wrapper col-sm-4 col-md-3 col-lg-2 col-6\\"><a href=\\"#/note-detail/1583267092854\\" class=\\"note-card v-card v-card--link v-sheet theme--light lt-white\\" tabindex=\\"0\\">
      <div class=\\"v-image v-responsive theme--light\\" style=\\"height: 100px;\\">
        <transition-stub name=\\"fade-transition\\" mode=\\"in-out\\">
          <div class=\\"v-image__image v-image__image--preload v-image__image--cover\\" style=\\"background-position: center center;\\"></div>
        </transition-stub>
        <div class=\\"v-responsive__content\\"></div>
      </div>
      <div class=\\"v-card__text main-text\\">
        <p>Note 1583267092854 </p>
      </div>
      <div class=\\"spacer\\"></div>
      <div class=\\"v-card__text footer-text\\"><span>
      2020-03-03
    </span>
        <!---->
      </div>
    </a></div>
  <div class=\\"note-wrapper col-sm-4 col-md-3 col-lg-2 col-6\\"><a href=\\"#/note-detail/1583267133077\\" class=\\"note-card v-card v-card--link v-sheet theme--light lt-purple\\" tabindex=\\"0\\">
      <!---->
      <div class=\\"v-card__text main-text\\">
        <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras ac justo scelerisque, accumsan ...</p>
      </div>
      <div class=\\"spacer\\"></div>
      <div class=\\"v-card__text footer-text\\"><span>
      2020-03-03
    </span>
        <!---->
      </div>
    </a></div>
  <div class=\\"note-wrapper col-sm-4 col-md-3 col-lg-2 col-6\\"><a href=\\"#/note-detail/1583267075908\\" class=\\"note-card v-card v-card--link v-sheet theme--light lt-red\\" tabindex=\\"0\\">
      <!---->
      <div class=\\"v-card__text main-text\\">
        <p>Note 1583267075908</p>
      </div>
      <div class=\\"spacer\\"></div>
      <div class=\\"v-card__text footer-text\\"><span>
      2020-03-03
    </span>
        <!---->
      </div>
    </a></div>
  <div class=\\"note-wrapper col-sm-4 col-md-3 col-lg-2 col-6\\"><a href=\\"#/note-detail/1583266765730\\" class=\\"note-card v-card v-card--link v-sheet theme--light lt-green\\" tabindex=\\"0\\">
      <!---->
      <div class=\\"v-card__text main-text\\">
        <p>Pellentesque sit amet felis felis. Fusce eu orci tempus metus dictum imperdiet. In consectet ...</p>
      </div>
      <div class=\\"spacer\\"></div>
      <div class=\\"v-card__text footer-text\\"><span>
      2020-03-03
    </span>
        <!---->
      </div>
    </a></div>
  <div class=\\"note-wrapper col-sm-4 col-md-3 col-lg-2 col-6\\"><a href=\\"#/note-detail/1583267808194\\" class=\\"note-card v-card v-card--link v-sheet theme--light lt-yellow\\" tabindex=\\"0\\">
      <!---->
      <div class=\\"v-card__text main-text\\">
        <p>Donec at nibh ut risus vehicula luctus ac ut magna. Donec lobortis purus sed leo convallis, ...</p>
      </div>
      <div class=\\"spacer\\"></div>
      <div class=\\"v-card__text footer-text\\"><span>
      2020-03-03
    </span>
        <!---->
      </div>
    </a></div>
  <div class=\\"note-wrapper col-sm-4 col-md-3 col-lg-2 col-6\\"><a href=\\"#/note-detail/1583311360827\\" class=\\"note-card v-card v-card--link v-sheet theme--light lt-white\\" tabindex=\\"0\\">
      <!---->
      <div class=\\"v-card__text main-text\\">
        <p>Morbi ultrices et purus quis sagittis. Sed justo libero, lobortis in ante sed, semper rhoncu ...</p>
      </div>
      <div class=\\"spacer\\"></div>
      <div class=\\"v-card__text footer-text\\"><span>
      2020-03-04
    </span>
        <!---->
      </div>
    </a></div>
  <div class=\\"note-wrapper col-sm-4 col-md-3 col-lg-2 col-6\\"><a href=\\"#/note-detail/1583311794631\\" class=\\"note-card v-card v-card--link v-sheet theme--light lt-yellow\\" tabindex=\\"0\\">
      <!---->
      <div class=\\"v-card__text main-text\\">
        <p>Interdum et malesuada fames ac ante ipsum primis in faucibus. Nulla facilisi. Fusce lobortis ...</p>
      </div>
      <div class=\\"spacer\\"></div>
      <div class=\\"v-card__text footer-text\\"><span>
      2020-03-04
    </span>
        <!---->
      </div>
    </a></div>
  <div class=\\"note-wrapper col-sm-4 col-md-3 col-lg-2 col-6\\"><a href=\\"#/note-detail/1583313433510\\" class=\\"note-card v-card v-card--link v-sheet theme--light lt-yellow\\" tabindex=\\"0\\">
      <!---->
      <div class=\\"v-card__text main-text\\">
        <p>Morbi finibus lacinia tortor quis imperdiet. Nam convallis arcu a tellus fringilla interdum. ...</p>
      </div>
      <div class=\\"spacer\\"></div>
      <div class=\\"v-card__text footer-text\\"><span>
      2020-03-04
    </span>
        <!---->
      </div>
    </a></div>
  <div class=\\"note-wrapper col-sm-4 col-md-3 col-lg-2 col-6\\"><a href=\\"#/note-detail/1583474432437\\" class=\\"note-card v-card v-card--link v-sheet theme--light lt-green\\" tabindex=\\"0\\">
      <!---->
      <div class=\\"v-card__text main-text\\">
        <p>Aenean at arcu sed tortor dignissim porta. Nullam eu risus mauris. Mauris et congue ipsum. Q ...</p>
      </div>
      <div class=\\"spacer\\"></div>
      <div class=\\"v-card__text footer-text\\"><span>
      2020-03-06
    </span>
        <!---->
      </div>
    </a></div>
  <div class=\\"note-wrapper col-sm-4 col-md-3 col-lg-2 col-6\\"><a href=\\"#/note-detail/1583478409029\\" class=\\"note-card v-card v-card--link v-sheet theme--light lt-blue\\" tabindex=\\"0\\">
      <!---->
      <div class=\\"v-card__text main-text\\">
        <p>Fusce quis risus tempus, viverra ante et, pellentesque elit. Aliquam nec odio blandit, solli ...</p>
      </div>
      <div class=\\"spacer\\"></div>
      <div class=\\"v-card__text footer-text\\"><span>
      2020-03-06
    </span>
        <!---->
      </div>
    </a></div>
  <div class=\\"note-wrapper col-sm-4 col-md-3 col-lg-2 col-6\\"><a href=\\"#/note-detail/1583478644306\\" class=\\"note-card v-card v-card--link v-sheet theme--light lt-yellow\\" tabindex=\\"0\\">
      <!---->
      <div class=\\"v-card__text main-text\\">
        <p>Quisque sed mollis mauris, ut fermentum mi. Suspendisse vitae nulla ex. Nunc ligula orci, ia ...</p>
      </div>
      <div class=\\"spacer\\"></div>
      <div class=\\"v-card__text footer-text\\"><span>
      2020-03-06
    </span>
        <!---->
      </div>
    </a></div>
  <div class=\\"note-wrapper col-sm-4 col-md-3 col-lg-2 col-6\\"><a href=\\"#/note-detail/1583311559736\\" class=\\"note-card v-card v-card--link v-sheet theme--light lt-blue\\" tabindex=\\"0\\">
      <!---->
      <div class=\\"v-card__text main-text\\">
        <p>Nulla elementum sapien in sapien facilisis, nec porttitor tellus laoreet. Cras condimentum r ...</p>
      </div>
      <div class=\\"spacer\\"></div>
      <div class=\\"v-card__text footer-text\\"><span>
      2020-03-08
    </span>
        <!---->
      </div>
    </a></div>
</div>"
`;
